parameter/BinaryStream.cpp

```diff
@@ -151,33 +149,40 @@ void CBinaryStream::reset()
     _uiPos = 0;
 }
 
+void CBinaryStream::assertValidAccess(size_t offset, size_t size) const
+{
+	assert(offset + size <= getSize());
+}
+
 void CBinaryStream::write(const uint8_t* puiData, size_t uiSize)
 {
-    assert(_uiPos + uiSize <= _uiDataSize);
+    assertValidAccess(_uiPos, uiSize);
```

Comment:
std::copy will throw an exception in case of invalid access.
assertValidAcess looks duplicate here

Throws if either an element assignment or an operation on iterators throws.
Note that invalid arguments cause undefined behavior.